doc/user/content/transform-data/temporal-filters.md

```diff
@@ -0,0 +1,368 @@
+This query determines the top ten unshipped orders by value as of some date (here: `'1995-03-15'`). Wouldn't it be neat to instead monitor it for all dates, continually, as it happens?
+
+All we need to do is change those two lines with the `'1995-03-15'` in them.
+
+```sql
+...
+    AND o_orderdate < mz_now()
+    AND l_shipdate > mz_now()
+...
+```
+
+That was an easy change to type. Let's discuss what these new bounds do.
+
+The first changed constraint restricts our attention to orders placed before "now", which has the effect of keeping orders out of the query until we've reached their `o_orderdate` column. In a real setting, this is probably a bit of a no-op, in that the record probably lands in our input stream around the order date anyhow, and doesn't need to be suppressed before then.
+
+The second changed constraint restricts our attention to `lineitem` records that have not shipped by "now". This has the effect of deleting the record once we reach `l_shipdate`, effectively garbage collecting that relation for us, which is especially nice as `lineitem` is the largest of the input relations (it is a "fact table"), and which churns the most.
+
+Expressed this way, with temporal filters, the memory footprint of the view will be proportional to the sizes of `orders` and `customer`, plus as much of `lineitem` is present but has not yet shipped. If we wanted to tighten our belt even more, we could add a further constraint that we aren't looking at orders that are too old
+
+```sql
+    AND o_orderdate + '90 days' > mz_now()
+```
+
+This will collect up the `orders` relation in addition to `lineitem`, and also prevent us from always seeing that one order from two years back that never shipped.
+
+## Conclusions
+
+Temporal filters are pretty neat stuff. I hope you are half as excited as I am.
+
+There are some limitations. I should have mentioned this earlier.
+
+You can only use `mz_now()` in `WHERE` clauses, where it must be directly compared to expressions not containing `mz_now()`, or in a conjunction (`AND`) with other clauses like that. You aren't allowed to use `!=` at the moment, but clever folks could figure out how to fake that out. For the reasoning on all this, check out the implementation discussion next!
+
+Limitations notwithstanding, I'm personally very excited about these temporal filters. They open up the door to functionality and behaviors that streaming systems only provide through special language extensions. But, all you really need is SQL, and the ability to refer to time, to make your data run.
+
+## Appendex: Implementation
+
+Some of you are surely here to hear how the magic works.
+
+The magic lives in `filter.rs`, which is the Rust source for our filter operator. Normally, the filter logic is very simple, and evaluates predicates against records and drops those records that do not pass the predicate. That code was sufficiently simple that it did not previously merit its own file (it was 10 lines of code, roughly).
```

Comment:
`filter.rs` probably doesn't exist any more, and is probably out of scope. It's in `linear.rs` now, if we want to keep the text!